corehq/apps/export/models/deid_export.py

```diff
@@ -12,8 +12,10 @@ class DeIdHash(models.Model):
     domain = models.TextField(max_length=255, db_index=True)
 
     @classmethod
-    def get_deid(cls, value, doc):
-        domain = doc['domain'] if isinstance(doc, dict) else doc.domain
+    def get_deid(cls, value, doc, domain=None):
+        domain = (domain if domain is not None
```

Comment:
In digging into behavior with de-identifying data in repeat groups on form exports, I realized that the `doc` passed in for these fields is just a dict representation of the repeat itself, not an entire form or case as is typical, so they would have errored on trying to access `doc['domain']`. I'm passing in domain directly here now in most situations, except where it's used by the `CaseCopier`, because that should always have a `CommCareCase` to pass in.

That probably means I can remove the following line, but I sort of lean toward keeping in in to catch edge cases, because there's definitely funky data in some places in HQ.
```
else doc['domain'] if isinstance(doc, dict)
```